campaigns.py

```diff
@@ -32,26 +39,73 @@ def homepage_template():
     <a href="{{ more_info|e }}">More information</a>
   </div>
 </div>""")
-    env['template_contents'] = template.render(context)
 
-    print "Template contents:\n%s" % env['template_contents']
+    env['frontend_template_contents'] = template.render(env.context)
+    print "Template contents:\n%s" % env['frontend_template_contents']
+
+
+@runs_once
+def static_template():
+    """Promotes a campaign to GOV.UK static pages"""
+    template = Template("""<p>{{ heading|e }}<br />
+    {{ extra_info|e }}</p>
+    <a href="#" class="right">{{ more_info|e }}</a>""")
+
+    env['static_template_contents'] = template.render(env.context)
+    print "Template contents:\n%s" % env['static_template_contents']
+
+
+@task
+@roles('class-frontend')
+def deploy_to_frontend():
+    execute(set_context)
+    execute(frontend_template)
+    remote_filename = '/var/apps/frontend/app/views/root/_campaign_notification.html.erb'
+    put(StringIO.StringIO(env['frontend_template_contents']), remote_filename, use_sudo=True, mirror_local_mode=True)
+    sudo('chown deploy:deploy %s' % remote_filename)
+    execute(app.reload, 'frontend')
+
+@task
+@roles('class-frontend')
+def deploy_to_static():
+    execute(set_context)
+    execute(static_template)
+    remote_filename = "/var/apps/static/app/views/notifications/banner_{}.erb".format(env.campaign_class)
+    temp_filename = "/tmp/banner_{}.erb".format(env.campaign_class)
+    put(StringIO.StringIO(env['static_template_contents']), temp_filename, use_sudo=True, mirror_local_mode=True)
+    sudo("mv {} {}".format(temp_filename,remote_filename))
+    sudo('chown deploy:deploy %s' % remote_filename)
+    execute(app.reload, 'static')
 
 @task
 @roles('class-frontend')
-def deploy_to_homepage():
-    execute(homepage_template)
+@runs_once
+def deploy_to_all():
+    execute(deploy_to_frontend)
+    execute(deploy_to_static)
 
-    contents = env['template_contents']
 
+@task
+@roles('class-frontend')
+def remove_from_frontend():
+    execute(set_context)
```

Comment:
@mattbostock Good question. I should likely have put something in a commit message regarding this. Frontend and Static deal with these banner notifications differently. [Frontend](https://github.com/alphagov/frontend/blob/master/app/views/root/_campaign_notification.html.erb.example#L4) deals with them by setting the `campaign_class` in the notification erb template. Simples. [Static](https://github.com/alphagov/static/tree/master/app/views/notifications) however has file names for each `campaign_class`, which means to remove the banner from Static, you need to know the `campaign_class` to write a blank file to. It doesn't seem very logical to need to `set_context` to _remove_ some text but this was the simplest solution I came up with. It is documented in the [opsmanual](https://github.gds/gds/opsmanual/pull/546/files#diff-454ae838cb3b8bb1ea1ce9886032ba72R98)

I could add another method just for removing which takes the `campaign_class` as a prompt, perhaps this would be more obvious. Open to suggestions.